x-pack/plugins/ml/public/components/anomalies_table/anomalies_table.js

```diff
@@ -0,0 +1,402 @@
+      },
+      sortable: true
+    });
+
+    // Assume that if we are showing typical, there will be an actual too,
+    // so we can add a column to describe how actual compares to typical.
+    const nonTimeOfDayOrWeek = items.some((item) => {
+      const summaryRecFunc = item.source.function;
+      return summaryRecFunc !== 'time_of_day' && summaryRecFunc !== 'time_of_week';
+    });
+    if (nonTimeOfDayOrWeek === true) {
+      columns.push({
+        field: 'metricDescriptionSort',
+        name: 'description',
+        render: (metricDescriptionSort, item) => {
+          return (
+            <DescriptionCell
+              actual={item.actual}
+              typical={item.typical}
+            />
+          );
+        },
+        sortable: true
+      });
+    }
+  }
+
+  columns.push({
+    field: 'jobId',
+    name: 'job ID',
+    sortable: true
+  });
+
+  const showExamples = items.some(item => item.entityName === 'mlcategory');
+  const showLinks = showViewSeriesLink === true && items.some((item) => showLinksMenuForItem(item));
+
+  if (showLinks === true) {
+    columns.push({
+      name: 'links',
+      render: (item) => {
+        if (showLinksMenuForItem(item) === true) {
+          return (
+            <LinksMenu
+              anomaly={item}
+              showViewSeriesLink={showViewSeriesLink}
+              isAggregatedData={isAggregatedData}
+              interval={interval}
+              timefilter={timefilter}
+            />
+          );
+        } else {
+          return null;
+        }
+      },
+      sortable: false
+    });
+  }
+
+  if (showExamples === true) {
+    columns.push({
+      name: 'category examples',
+      sortable: false,
+      truncateText: true,
+      render: (item) => {
+        const examples = _.get(examplesByJobId, [item.jobId, item.entityValue], []);
+        return (
+          <EuiText size="xs">
+            {examples.map((example, i) => {
+              return <span key={`example${i}`} className="category-example">{example}</span>;
+            }
+            )}
+          </EuiText>
+        );
+      }
+    });
+  }
+
+  return columns;
+}
+
+class AnomaliesTable extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      itemIdToExpandedRowMap: {}
+    };
+  }
+
+  isShowingAggregatedData = () => {
+    return (this.props.tableData.interval !== 'second');
+  };
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    // Update the itemIdToExpandedRowMap state if a change to the table data has resulted
+    // in an anomaly that was previously expanded no longer being in the data.
+    const itemIdToExpandedRowMap = prevState.itemIdToExpandedRowMap;
+    const prevExpandedNotInData = Object.keys(itemIdToExpandedRowMap).find((rowId) => {
+      const matching = nextProps.tableData.anomalies.find((anomaly) => {
+        return anomaly.rowId === rowId;
+      });
+
+      return matching === undefined;
```

Comment:
nit, similarly to ternary expressions, we should wrap conditions like this.
i.e. `(matching === undefined)`